4-tools/sieve.py

```diff
@@ -1,4 +1,4 @@
-@profile
+@profile  # noqa: F821, D100
```

Comment:
[D100](https://docs.astral.sh/ruff/rules/undocumented-public-module/) is about module docstrings, so it should be ignored at the module level.
```suggestion
# ruff: noqa: D100

@profile  # noqa: F821
```
But how can this code do anything useful if `profile()` is not defined?

Reply:
`@profile` is not a function but a decorator for line_profiler.